sky/backends/backend_utils.py

```diff
@@ -990,6 +1000,115 @@ def get_ready_nodes_counts(pattern, output):
     return ready_head, ready_workers
 
 
+@timeline.event
+def _deterministic_yaml_hash(yaml_path: str) -> str:
```

Comment:
A thought here is that, the optimization here is optional, so if there are any errors with getting the hash of the config, should we just skip the optimization and let it continue?

Reply:
added exception handling to the caller (`write_cluster_config`)